### PR TITLE
Sync Dependabot Secrets

### PR DESCRIPTION
**Pull request by kshitijzutshi222**
### Issue

At present the manage-secrets workflow, that uses Repo-Secret-Manager action doesn't sync secrets for dependabot. User should have the choice to sync/not sync dependabot secrets from ix-tools

### Changes Made 

1. action.yml changes to incorporate new input field
2. Modification to entrypoint.sh for the new input
3. main.py file changes, new functions added for - public key for repo retrieval, encrypt secret, add dependabot secret, update dependabot secret, remove dependabot secret

### CODE REVIEW

**Review by rdconnelly — APPROVED** [review]

File: action.yml [hunk 1]
```diff
@@ -23,6 +23,10 @@ inputs:
     description: 'A specific repo to sync the secrets to'
     mandatory: False
     default: ''
+  dependabotSecretsSync:
+    description: 'Sync secrets to dependabot (yes/no), default yes'
+    mandatory: False
+    default: 'no'
```
Comment (sokunna) [inline action.yml:29]: this is different from the description above?

File: main.py [hunk 2]
```diff
@@ -150,6 +171,74 @@ def add_secret(token, target_repository, secret_name, secret_value):
     else:
         print(f"Secret \"{secret_name}\" already exists in {repo_name}")
 
+def add_dependabot_secret(token, target_repository, secret_name, secret_value):
+    repo_name = target_repository.name
+    repo_owner = "philips-internal"
```
Comment (sokunna) [inline main.py:176]: i don't think this should be hardcoded here

File: main.py [hunk 3]
```diff
@@ -150,6 +171,74 @@ def add_secret(token, target_repository, secret_name, secret_value):
...
+        if response.status_code == 201:
+            print(f"dependabot Secret \"{secret_name}\" added to {repo_name}")
+        else:
+            print(f"dependabot Secret \"{secret_name}\" could NOT be added to {repo_name}")
+    else:
+        print(f"dependabot Secret \"{secret_name}\" already exists in {repo_name}")
+
+def update_dependabot_secret(token, target_repository, secret_name, secret_value):
+    repo_name = target_repository.name
+    repo_owner = "philips-internal"
```
Comment (sokunna) [inline main.py:205]: ditto

File: main.py [hunk 4]
```diff
@@ -150,6 +171,74 @@ def add_secret(token, target_repository, secret_name, secret_value):
...
+        response = requests.patch(url, headers=headers, data=json.dumps(data))
+        print(f"Response Code: {response.status_code}")
+        if response.status_code == 204:
+            print(f"dependabot Secret \"{secret_name}\" updated in {repo_name}")
+        else:
+            print(f"dependabot Secret \"{secret_name}\" could NOT be updated in {repo_name}")
+
+def delete_dependabot_secret(token, target_repository, secret_name):
+    repo_name = target_repository.name
+    repo_owner = "philips-internal"
```
Comment (sokunna) [inline main.py:232]: ditto

File: main.py (hunk 2 above)
Comment (kshitijzutshi222) [reply main.py]: fixed.

File: action.yml [hunk 5]
```diff
@@ -23,6 +23,14 @@ inputs:
...
+  repo_owner:
+    description: 'Repository Owner Name'
+    mandatory: False
+    default: 'philips-internal'
```
Comment (sokunna) [inline action.yml:33]: I would not use this as a default... the value should be passed

File: action.yml (hunk 1 above)
Comment (sokunna) [reply action.yml:29]: the description says Yes, the default says No

Comment (kshitijzutshi222) [reply action.yml:29]: Done

File: action.yml (hunk 5 above)
Comment (kshitijzutshi222) [reply action.yml:33]: The action will take the repo owner name as input from the calling workflow, As I needed default to be a non null value, I  assigned its value as philips-internal.

Comment (kshitijzutshi222) [reply action.yml:33]: So this -  the value should be passed

I dont understand clearly.

Comment (kshitijzutshi222) [reply action.yml:33]: Ok in that case, I will make this mandatory and hence remove the default param

Comment (kshitijzutshi222) [reply action.yml]: default - "org-name"

File: main.py [hunk 6]
```diff
@@ -173,16 +267,43 @@ def add_secret(token, target_repository, secret_name, secret_value):
...
-                        repo.delete_secret(inp.secret_names[i])
-                        print(f"Secret \"{inp.secret_names[i]}\" removed from {repo.name}")
-                except UnknownObjectException:
-                    print(f"The provided token does not have permission to manage {repo.name}, it is being skipped")
+            if len(inp.dependabotSecretsSync) == 0 or inp.dependabotSecretsSync.lower() == "yes" or inp.dependabotSecretsSync.lower() == '':
+                if not inp.interactive or apply_action(repo.name):
+                    try:
+                        if inp.action == createCommand:
+                            add_secret(inp.token, repo, inp.secret_names[i], inp.secret_values[i])
+                            if not inp.repoOwner == "org-name":
```
Comment (gauravngandhi) [inline main.py:275]: create a const for "org-name" as well

**Review by gauravngandhi — CHANGES_REQUESTED** [review]

**Review by gauravngandhi — APPROVED** [review]
LGTM

**Review by sokunna — APPROVED** [review]

**Pull request merged**